core/types/block.go

```diff
@@ -99,8 +105,18 @@ func (h *Header) SanityCheck() error {
 
 // EmptyBody returns true if there is no additional 'body' to complete the header
 // that is: no transactions.
-func (h *Header) EmptyBody() bool {
-	return h.TxHash == EmptyRootHash
+func (h *Header) EmptyBody() (bool, error) {
+	extra, err := h.IstanbulExtra()
+
+	if err != nil {
+		return false, err
+	}
+
+	if extra != nil {
+		return false, nil
+	}
```

Comment:
@piersy I'm nor sure about this part. `h.IstanbulExtra` will always return either an `extra` or an `err`, so we now can never reach the statement in line 119, which doesn't make much sense.

Reply:
Done